client/app/api/image-process/route.ts

```diff
@@ -0,0 +1,23 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { getPlaiceholder } from 'plaiceholder';
+
+export async function GET(req: NextRequest, res: NextResponse) {
```

Comment:
In the documentation, it says that for blur preload images, I need to provide an encoded version of the image, and they recommend using a plaiceholder lib.

https://nextjs.org/docs/pages/api-reference/components/image#blurdataurl

So instead of making another API call to the S3 bucket and getting the base64 version of the image every time, I just encoded it while creating the recipe to save it in the database and access it easily.

I was struggling to make it, but I think it provides a good UI so users don't experience crazy layout shifts while loading the page.